lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -115,6 +126,51 @@ public function iterate($stmt, $resultSetMapping, array $hints = [])
         return new IterableResult($this);
     }
 
+    /**
+     * Initiates a row-by-row hydration.
+     *
+     * @param mixed[] $hints
+     *
+     * @return iterable<mixed>
+     */
+    public function getIterable(
+        Statement $stmt,
+        ResultSetMapping $resultSetMapping,
+        array $hints = []
+    ) : iterable {
+        $this->stmt  = $stmt;
+        $this->rsm   = $resultSetMapping;
+        $this->hints = $hints;
+
+        $evm = $this->em->getEventManager();
+
+        $evm->addEventListener([Events::onClear], $this);
+
+        $this->prepare();
+
+        $result = [];
+
+        while (true) {
+            $row = $this->stmt->fetch(FetchMode::ASSOCIATIVE);
+
+            if ($row === false || $row === null) {
+                $this->cleanup();
+
+                break;
+            }
+
+            $this->hydrateRowData($row, $result);
+
+            if (is_array($result)) {
```

Comment:
I don't understand why is this check here. When is it not an array?

Reply:
Hard to say with all these references of what's supposed to be an array. In the `ArrayHydrator` there's also this:

https://github.com/doctrine/orm/blob/6dc46e78ccf4fe75c80884e7abe41b811f2204b6/lib/Doctrine/ORM/Internal/Hydration/ArrayHydrator.php#L253-L255

and code like `$this->resultPointers[$dqlAlias] =& $coll;`

It's hard to tell if `null` or anything else would be possible.

Reply:
I ran test suite and it was always an array 🤔

Reply:
Yeah let's go with assumption it's always array. That's the only way how to verify this.